sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java

```diff
@@ -2025,13 +2026,9 @@ public Mono<FileListResponse> listFiles(FilePurpose purpose) {
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return a list of previously uploaded files on successful completion of {@link Mono}.
      */
-    @Generated
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<FileListResponse> listFiles() {
-        // Generated convenience method for listFilesWithResponse
-        RequestOptions requestOptions = new RequestOptions();
-        return listFilesWithResponse(requestOptions).flatMap(FluxUtil::toMono)
-            .map(protocolMethodData -> protocolMethodData.toObject(FileListResponse.class));
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public Flux<OpenAIFile> listFiles() {
```

Comment:
I am a bit split here, since `Flux` is generally the iterable type to use, if each item in the list was fetched asynchronously, but in this case the request returns a bulk of `List<OpenAIFile>`s which is best represented by a `Mono`. 

I would also prefer keep this looking the same as we have over in the Assistants SDK which is `Mono<List<OpenAIFile>>` .

If this change is product of customer feedback, then let's have a further discussion as this would constitute a significant change in the method signature, having a different return type.

Reply:
Yeah. Let's stay the same pattern and leave it to future API design discussions